pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountHLLPlusAggregationFunction.java

```diff
@@ -0,0 +1,469 @@
+import org.apache.pinot.core.query.aggregation.groupby.GroupByResultHolder;
+import org.apache.pinot.core.query.aggregation.groupby.ObjectGroupByResultHolder;
+import org.apache.pinot.segment.spi.AggregationFunctionType;
+import org.apache.pinot.segment.spi.index.reader.Dictionary;
+import org.apache.pinot.spi.data.FieldSpec.DataType;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.roaringbitmap.PeekableIntIterator;
+import org.roaringbitmap.RoaringBitmap;
+
+
+public class DistinctCountHLLPlusAggregationFunction extends BaseSingleInputAggregationFunction<HyperLogLogPlus, Long> {
+  // The parameter "p" determines the precision of the sparse list in HyperLogLogPlus.
+  protected final int _p;
+  // The "sp" parameter specifies the number of standard deviations that the sparse list's precision should be set to.
+  protected final int _sp;
+
+  public DistinctCountHLLPlusAggregationFunction(List<ExpressionContext> arguments) {
+    super(arguments.get(0));
+    int numExpressions = arguments.size();
+    // This function expects 1 or 2 or 3 arguments.
+    Preconditions.checkArgument(numExpressions <= 3, "DistinctCountHLLPlus expects 2 or 3 arguments, got: %s",
+        numExpressions);
+    if (arguments.size() == 2) {
+      _p = arguments.get(1).getLiteral().getIntValue();
+      _sp = CommonConstants.Helix.DEFAULT_HYPERLOGLOG_PLUS_SP;
+    } else if (arguments.size() == 3) {
+      _p = arguments.get(1).getLiteral().getIntValue();
+      _sp = arguments.get(2).getLiteral().getIntValue();
+    } else {
+      _p = CommonConstants.Helix.DEFAULT_HYPERLOGLOG_PLUS_P;
+      _sp = CommonConstants.Helix.DEFAULT_HYPERLOGLOG_PLUS_SP;
+    }
+  }
+
+  public int getP() {
+    return _p;
+  }
+
+  public int getSp() {
+    return _sp;
+  }
+
+  @Override
+  public AggregationFunctionType getType() {
+    return AggregationFunctionType.DISTINCTCOUNTHLLPLUS;
+  }
+
+  @Override
+  public AggregationResultHolder createAggregationResultHolder() {
+    return new ObjectAggregationResultHolder();
+  }
+
+  @Override
+  public GroupByResultHolder createGroupByResultHolder(int initialCapacity, int maxCapacity) {
+    return new ObjectGroupByResultHolder(initialCapacity, maxCapacity);
+  }
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    BlockValSet blockValSet = blockValSetMap.get(_expression);
+
+    // Treat BYTES value as serialized HyperLogLogPlus
+    DataType storedType = blockValSet.getValueType().getStoredType();
+    if (storedType == DataType.BYTES) {
+      byte[][] bytesValues = blockValSet.getBytesValuesSV();
+      try {
+        HyperLogLogPlus hyperLogLogPlus = aggregationResultHolder.getResult();
+        if (hyperLogLogPlus == null) {
+          hyperLogLogPlus = ObjectSerDeUtils.HYPER_LOG_LOG_PLUS_SER_DE.deserialize(bytesValues[0]);
+          aggregationResultHolder.setValue(hyperLogLogPlus);
+        }
```

Comment:
We need the else block if the check is not qualified:
```
} else {
    hyperLogLogPlus.addAll(ObjectSerDeUtils.HYPER_LOG_LOG_PLUS_SER_DE.deserialize(bytesValues[0]));
}
```

Reply:
got your point, updated, thanks!